### ISSUE TITLE
when will Redis 7 for windows release?

### ISSUE THREAD

**Issue by debugCode1024** [tporadowski/redis#144]
I am looking forward redis 7 for windows, Thanks

**Comment by btx638** [comment]
found a 7.0.2 x64 release ， but not test 
https://github.com/zkteco-home/redis-windows

**Comment by superlbr** [comment]
according to this article,just build from source with msys2. https://blog.csdn.net/xuejianxinokok/article/details/125423154

**Comment by thilux** [comment]
Is there a procedure/guide on what changes need to be done so that others can help and create PRs for Redis 6.2.X and 7.X? I am interested in helping.

**Comment by tporadowski** [comment]
@thilux I appreciate your offer to help! This fork is a continuation of work done by Microsoft OpenTech team. This is a tedious process, but maybe I could not see a more efficient way to do this so far :). There are some explanations in header files why `PORT_...` macros for various data types are used in this fork, and there are parts of code that are rewritten to Windows-specific alternatives (Win IOCP, `fork()` implementation, etc.). What I was doing so far was to take a given release branch from original Redis repo, and then using WinMerge tool - going through all the differences file by file. Then for each file those types must be changed, any new code/changes taken over, changes to Windows-specific parts must be carefully examined and applied (they're usually in `#if ... #else ... #endif` blocks). Then when it compiles - tests must be run, any issues corrected, and again tested.

**Comment by tobaba** [comment]
> found a 7.0.2 x64 release ， but not test https://github.com/zkteco-home/redis-windows

Good ,Thank you.

**Comment by ykexc** [comment]
> 

is it available?

**Comment by liudonghua123** [comment]
> found a 7.0.2 x64 release ， but not test https://github.com/zkteco-home/redis-windows

It's not open source project.

**Comment by jonahbeckford** [comment]
I have Windows binaries for Redis 7 available on https://gitlab.com/diskuv-ocaml/distributions/dkml-redis. Be aware they are compiled with GCC so they are not truly native like tporadowski/redis.

**Comment by liudonghua123** [comment]
Recently, I found another repo https://github.com/redis-windows/redis-windows which is build with mingw and use github action.

**Comment by zkteco-home** [comment]
> Recently, I found another repo https://github.com/redis-windows/redis-windows which is build with mingw and use github action.

How can you know it is compliled with mingw?I never use mingw or cygwin,it is compiled with VS2022,real native binary files

**Comment by zkteco-home** [comment]
> > 
> 
> is it available?

so far,it is the most stable windows edition

**Comment by liudonghua123** [comment]
> > Recently, I found another repo https://github.com/redis-windows/redis-windows which is build with mingw and use github action.
> 
> How can you know it is compliled with mingw?I never use mingw or cygwin,it is compiled with VS2022,real native binary files

I didn't mean your repo, I said https://github.com/redis-windows/redis-windows.
It's true open sourced, the main build script is on
https://github.com/redis-windows/redis-windows/blob/main/.github/workflows/ci-diy.yml

**Comment by zkteco-home** [comment]
sorry for that,that is not my repo.i did a test ,with mingw,it is at least slower 40% then native compile,this is result:
with Mingw(redis-windows/redis-windows):

![image](https://user-images.githubusercontent.com/66519168/233817237-fb1c11d3-c526-4d6a-abcc-e115dfe47737.png)

with VS2022(zkteco-home/redis-windows:

![image](https://user-images.githubusercontent.com/66519168/233817226-6de28198-37a7-481a-a534-ac3eeefd57d4.png)

**Comment by jhajjaarap** [comment]
> It's true open sourced, ...

Yes, this is most important.

**Comment by dblife1024** [comment]
Yes, Redis7 is most important，I am looking forward redis 7 for windows, Thanks

**Comment by zkteco-home** [comment]
> Yes, Redis7 is most important，I am looking forward redis 7 for windows, Thanks

https://github.com/zkteco-home/redis-windows

**Comment by Lixuhuilll** [comment]
> > It's true open sourced, ...
> 
> Yes, this is most important.

Agree, I have no requirements for the performance of the Windows version, only for consistent functionality and complete open source.

**Comment by jfcherng** [comment]
> > > It's true open sourced, ...
> > 
> > 
> > Yes, this is most important.
> 
> Agree, I have no requirements for the performance of the Windows version, only for consistent functionality and complete open source.

https://github.com/redis-windows/redis-windows

**Comment by pwin32** [comment]
If anyone still interested here, I've made native windows port of version 6.2, 7.2, 8.8, and 8.10 based on 5.0 of this project, fully open source and all releases built, tested, and published by github actions ci.
No runtime dependency of cygwin or msys2 needed.
https://github.com/pwin32/redis/releases
